test/unit/math/fwd/mat/fun/log_mix_test.cpp

```diff
@@ -4,149 +4,165 @@
 
 using stan::math::fvar;
 using stan::math::log_mix;
-
-TEST(AgradFwdMatrixLogMix, vector_fd) {
```

Comment:
I wrote the original tests, and they weren't very comprehensive (or efficient). I extended them in this pull to better test all possible instantiations with less copy-and-paste repetition.

Reply:
This is good. I guess I'm just wondering how I should be verifying that the old behavior isn't changing with this new feature?

Reply:
By making the tests more comprehensive, the idea was that if the old behaviour changes then the tests would fail. Both versions tested the same things: whether the correct log-probability and partial derivatives were returned, the new version just tests it for more combinations of inputs. If the new feature caused the wrong values/derivatives to be returned for the other instantiations then the tests would fail.